### PR TITLE
fix-grant-save

### PR DESCRIPTION
**Pull request by EricWVGG**
This disables a thing that puts the form into perpetual react updates, and correctly returns true when the participants form is valid.

### CODE REVIEW

**Comment by EricWVGG** [comment]
edit: it now fixes an issue with creating a new grant.

**Comment by geometrizen** [comment]
I tested this locally, and on my first attempt, I could not get grant creation to work. In particular the glitchy text input (which I think is related to the infinite loop that you all were referring to yesterday) is still occurring for me-- I'll post some screen capture videos in slack. 

Here are some other general notes:

I don't think the admin / curator user should be allowed to enter text for the 'date of grant' -- not sure when this changed but it needs to be fixed. (Way too many things that could go wrong with a general admin user entering a datetime here)

typo: "Longline" should be "Logline" (if necessary, check [Nene's comment on this google doc](https://docs.google.com/document/d/1WlGZQut9EqNyP3IciuIJbatLgGTsJoHbNuCInCX20YA/edit#heading=h.60q6srg0l06g))

Love the form validation errors being displayed, thank you 👍 (I haven't checked that this is working for all relevant fields, but I saw it working for at least one)

So, I don't think this should be merged yet. Please let me know if I'm missing something. I'll try creating a grant again, typing more slowly

**Comment by geometrizen** [comment]
https://user-images.githubusercontent.com/102297373/210791663-cfc62f8a-dbaa-4ec2-a095-4470283d04cd.mov


https://user-images.githubusercontent.com/102297373/210791702-103c8529-fb77-4ac4-aa98-8e72c09e8590.mov

**Review by geometrizen — CHANGES_REQUESTED** [review]
Testing this locally did not work for me-- please see comment(s) in the general conversation of this PR.

**Comment by EricWVGG** [comment]
- The text capture bug still exists in the current codebase, and is lessened with this PR, so it’s not a regression.
- If “longline” is supposed to be “logline” then that’s a typo in Hasura. It should be fixed as a separate branch.
- “admin curator shouldn’t enter text for date of grant” I thought I explained that, apologies if I overlooked it: it is supposed to be a pre-fill read-only field, but the text capture bug was leading that to an infinite loop.

Since this does enable the creation of grants, and that’s holding up a lot of important work, and none of this PR is a regression, I would like it to be merged so we can move forward with testing and fixing other things (yes, including deeper investigation of the text capture glitch)

File: src/components/CreateGrants/NewGrantForm/index.tsx [hunk 1]
```diff
@@ -28,33 +28,36 @@ const NewGrantForm = () => {
...
-        keyword: '',
-        params: {},
-      })
-    }
-    setAdditionalErrors(errors)
-  }, [data])
+  //   useEffect(() => {
+  //     // set additional form validators here
+  //     const errors: Array<ErrorObject> = []
+  //     if (!validateLib.validate(data.project.walletAddress, 'ETH')) {
```
Comment (geometrizen) [inline src/components/CreateGrants/NewGrantForm/index.tsx:50]: are we validating the wallet address elsewhere? validating the wallet address seems like a good thing to do

File: src/components/CreateGrants/NewGrantForm/helpers/validators.ts [hunk 2]
```diff
@@ -36,4 +36,6 @@ export const validateProjectMembers = (projectMembers: Array<ProjectMember>) =>
     alert(usersWithIncorrectWallet.join())
     return false
   }
+
+  return true
```
Comment (geometrizen) [inline src/components/CreateGrants/NewGrantForm/helpers/validators.ts:40]: ah is this effectively where the wallet address validation is happening now?

**Review by geometrizen — APPROVED** [review]
looks okay and I won't block merging as I agree with Eric that we should test this on dev-- but I'm curious about my comments inline as well as the user-editability of the grant start date

File: src/components/CreateGrants/NewGrantForm/helpers/validators.ts (hunk 2 above)
Comment (EricWVGG) [reply src/components/CreateGrants/NewGrantForm/helpers/validators.ts:40]: Yes.

File: src/components/CreateGrants/NewGrantForm/index.tsx (hunk 1 above)
Comment (EricWVGG) [reply src/components/CreateGrants/NewGrantForm/index.tsx:50]: The validation is contributing to the text capture bug.

**Pull request merged**